pkg/controllers/node/ipam.go

```diff
@@ -137,7 +158,7 @@ func (c *NodeController) syncIPAMCleanup() error {
 
 			// Check to see if the pod still exists. If it does, then we shouldn't clean up
 			// this node, since it might come back online.
-			if pod != "" && c.podExists(pod, ns, node) {
+			if pod != "" && c.podExists(pod, ns, knode) {
```

Comment:
nit: Also check `knode != ""`

I think it kinda drops out in the podExists check, but I think it's a bit clearer to explicitly handle the "unknown knode" case.